### PR TITLE
Define the bucket hosting DSS event data

### PR DESCRIPTION
**Pull request by xbrianh**
This also defines the canonical interfaces to the event bucket and replica prefixes that will contain event data.

connected to #2293

### CODE REVIEW

File: environment [hunk 1]
```diff
@@ -66,6 +67,10 @@ DSS_S3_CHECKOUT_BUCKET_UNWRITABLE=org-hca-dss-checkout-unwritable
 DSS_S3_CHECKOUT_BUCKET_INTEGRATION=org-humancellatlas-dss-checkout-integration
 DSS_S3_CHECKOUT_BUCKET_STAGING=org-humancellatlas-dss-checkout-staging
 DSS_S3_CHECKOUT_BUCKET_PROD=org-humancellatlas-dss-checkout-prod
+DSS_FLASHFLOOD_BUCKET=org-humancellatlas-dss-events-dev
+DSS_FLASHFLOOD_BUCKET_INTEGRATION=org-humancellatlas-dss-events-integration
```
Comment (amarjandu) [inline environment]: Should these be set in the corresponding environment.{stage} file?

File: infra/buckets/s3.tf [hunk 2]
```diff
@@ -150,3 +150,12 @@ resource aws_s3_bucket dss_s3_checkout_bucket_unwritable {
 }
 POLICY
 }
+
+resource aws_s3_bucket dss_s3_events_bucket {
+  count = "${length(var.DSS_S3_BUCKET) > 0 ? 1 : 0}"
```
Comment (amarjandu) [inline infra/buckets/s3.tf]: does the var evaluated in the count need to be the event bucket?

File: environment (hunk 1 above)
Comment (xbrianh) [reply environment]: yep, fixed

File: infra/buckets/s3.tf (hunk 2 above)
Comment (xbrianh) [reply infra/buckets/s3.tf]: yes, fixed

File: infra/buckets/s3.tf [hunk 3]
```diff
@@ -150,3 +150,12 @@ resource aws_s3_bucket dss_s3_checkout_bucket_unwritable {
...
 POLICY
 }
+
+resource aws_s3_bucket dss_s3_events_bucket {
+  count = "${length(var.DSS_FLASHFLOOD_BUCKET) > 0 ? 1 : 0}"
+  bucket = "${var.DSS_FLASHFLOOD_BUCKET}"
+  server_side_encryption_configuration {
+    rule {apply_server_side_encryption_by_default {sse_algorithm = "AES256"}}
+  }
+  tags = "${local.common_tags}"
```
Comment (natanlao) [inline infra/buckets/s3.tf]: Is there a reason this isn't `tags = "${merge(local.common_tags, local.aws_tags)}"`?

File: tests/test_config.py [hunk 4]
```diff
@@ -43,6 +43,14 @@ def test_s3_checkout_bucket(self):
         Config.set_config(BucketConfig.TEST_FIXTURE)
         self.assertEquals(Config.get_s3_checkout_bucket(), os.environ["DSS_S3_CHECKOUT_BUCKET_TEST"])
 
+    def test_s3_events_bucket(self):
+        Config.set_config(BucketConfig.NORMAL)
+        self.assertEquals(Config.get_flashflood_bucket(), os.environ["DSS_FLASHFLOOD_BUCKET"])
```
Comment (natanlao) [inline tests/test_config.py]: Nit: `assertEquals` is deprecated, use `assertEqual` instead. (I was under the impression that `assertEquals` would stop working entirely, but it looks like it just raises a warning after 3.2, and in theory will stop working in the future.)

File: infra/buckets/s3.tf (hunk 3 above)
Comment (xbrianh) [reply infra/buckets/s3.tf]: fixed

File: tests/test_config.py (hunk 4 above)
Comment (xbrianh) [reply tests/test_config.py]: changed.

**Pull request merged**